extensions/binding/org.eclipse.smarthome.binding.mqtt/src/main/java/org/eclipse/smarthome/binding/mqtt/internal/discovery/NetworkDiscoveryService.java

```diff
@@ -48,21 +46,22 @@
  * @author David Graeff - Initial contribution
  */
 @Component(immediate = true, service = DiscoveryService.class, configurationPid = "discovery.networkmqttbroker")
+@NonNullByDefault
 public class NetworkDiscoveryService extends AbstractDiscoveryService {
     private static final int MAX_IPS_PER_INTERFACE = 255;
     private static final int TIMEOUT_IN_MS = 700;
 
     private final Logger logger = LoggerFactory.getLogger(NetworkDiscoveryService.class);
 
-    private Integer scannedIPcount;
+    private int scannedIPcount = 0;
 
     public NetworkDiscoveryService() {
         super(Collections.singleton(MqttBindingConstants.BRIDGE_TYPE_BROKER), 0);
     }
 
     @Override
     @Activate
-    protected void activate(Map<String, Object> config) {
+    protected void activate(@Nullable Map<String, @Nullable Object> config) {
```

Comment:
Why is this map nullable? Is this forced by some super declaration?

Reply:
I see this code a lot:
```java
protected final void activate(final Map<String, Object> config) {
    if (config != null) {
        properties = config;
    }
```

Apparently `config` can be null, but I haven't found the standard that specifies OSGIs `activated`.

Reply:
https://osgi.org/download/r4v42/r4.cmpn.pdf

> 112.5.8 Activate Method

> The activate method can take zero or
 more arguments. Each argument must 
be of one of the following types

AFAIK all of that arguments are non null.

> Map  – The component instance will be passed an unmodifiable Map containing the component properties.

Reply:
AbstractDiscoveryService declares it as `protected void activate(@Nullable Map<String, @Nullable Object> configProperties)`, so I need to leave it as it is for now.

Reply:
That's another story.
The activate method of the `AbstractDiscoveryService` is not called by the SCR but by the developers of the specific discovery implementation.
Sure, it could be non-null here, too but it is out of scope here (currently the homematic discovery service seems not be configurable and calls the activate of the abstract discovery service with an null argument).

Reply:
The problem is, that I'm using the same activate signature. I either hide/shadow the base implementation, which is a bad practise, or I need to match the exact same signature. What do you prefer?

Reply:
Big sorry, I am missed that you extend the `AbstractDiscoveryService` here. :innocent:
All fine...